xds/internal/resolver/xds_resolver_test.go

```diff
@@ -1459,557 +816,655 @@ func (s) TestResolverMaxStreamDuration(t *testing.T) {
+					},
 				},
-			},
-			selectErr: errUnsupportedClientRouteAction.Error(),
+			}),
 		},
+	}
+}
+
+func (s) TestXDSResolverHTTPFilters(t *testing.T) {
+	const methodName1 = "1"
+	const methodName2 = "2"
+	testFilterName := t.Name()
+
+	testCases := []struct {
+		name          string
+		listener      *v3listenerpb.Listener
+		rpcRes        map[string][][]string
+		wantStreamErr string
+		wantPathLen   int
+	}{
 		{
-			name: "NewStream error; ensure earlier interceptor Done is still called",
-			ldsFilters: []xdsresource.HTTPFilter{
-				{Name: "foo", Filter: &filterBuilder{path: &path}, Config: filterCfg{s: "foo1"}},
-				{Name: "bar", Filter: &filterBuilder{path: &path}, Config: filterCfg{s: "bar1", newStreamErr: errors.New("bar newstream err")}},
-				makeRouterFilter(t),
-			},
-			rtCfgUpdate: xdsresource.RouteConfigUpdate{
-				VirtualHosts: []*xdsresource.VirtualHost{
-					{
-						Domains: []string{targetStr},
-						Routes: []*xdsresource.Route{{
-							Prefix: newStringP("1"),
-							WeightedClusters: map[string]xdsresource.WeightedCluster{
-								"A": {Weight: 1},
-								"B": {Weight: 1},
-							},
-							ActionType: xdsresource.RouteActionRoute,
-						}},
-					},
+			name: "NewStream error - ensure earlier interceptor Done is still called",
+			listener: &v3listenerpb.Listener{
+				Name: defaultTestServiceName,
+				ApiListener: &v3listenerpb.ApiListener{
+					ApiListener: testutils.MarshalAny(t, &v3httppb.HttpConnectionManager{
+						RouteSpecifier: &v3httppb.HttpConnectionManager_RouteConfig{
+							RouteConfig: &v3routepb.RouteConfiguration{
+								Name: defaultTestRouteConfigName,
+								VirtualHosts: []*v3routepb.VirtualHost{{
+									Domains: []string{defaultTestServiceName},
+									Routes: []*v3routepb.Route{{
+										Match: &v3routepb.RouteMatch{
+											PathSpecifier: &v3routepb.RouteMatch_Prefix{Prefix: methodName1},
+										},
+										Action: &v3routepb.Route_Route{
+											Route: &v3routepb.RouteAction{
+												ClusterSpecifier: &v3routepb.RouteAction_WeightedClusters{
+													WeightedClusters: &v3routepb.WeightedCluster{
+														Clusters: []*v3routepb.WeightedCluster_ClusterWeight{
+															{Name: "A", Weight: wrapperspb.UInt32(1)},
+															{Name: "B", Weight: wrapperspb.UInt32(1)},
+														},
+													},
+												},
+											},
+										},
+									}},
+								}},
+							}},
+						HttpFilters: []*v3httppb.HttpFilter{
+							newHTTPFilter(t, "foo", testFilterName, "foo1", ""),
+							newHTTPFilter(t, "bar", testFilterName, "bar1", "bar newstream err"),
+							e2e.RouterHTTPFilter,
+						},
+					}),
 				},
 			},
 			rpcRes: map[string][][]string{
-				"1": {
-					{"build:foo1", "build:bar1", "newstream:foo1", "newstream:bar1" /* <err in bar1 NewStream> */, "done:foo1"},
+				methodName1: {
+					{"build:foo1", "build:bar1", "newstream:foo1", "newstream:bar1", "done:foo1"}, // err is bar1 NewStream
```

Comment:
This comment is cryptic. err in bar1 NewStream() maybe? Or just delete?

Reply:
Done.

Not sure how I ended up changing `in` to `is` :)